resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/DataprocWorkerChecker.scala

```diff
@@ -0,0 +1,67 @@
+package com.broadinstitute.dsp.resourceValidator
+
+import cats.effect.{Concurrent, Timer}
+import cats.mtl.Ask
+import cats.implicits._
+import com.broadinstitute.dsp.{
+  regionName,
+  resourceValidator,
+  CheckRunner,
+  CheckRunnerConfigs,
+  CheckRunnerDeps,
+  RuntimeCheckerDeps,
+  RuntimeWithWorkers
+}
+import io.chrisdavenport.log4cats.Logger
+import org.broadinstitute.dsde.workbench.google2.DataprocClusterName
+import org.broadinstitute.dsde.workbench.model.TraceId
+
+object DataprocWorkerChecker {
+  def impl[F[_]: Timer](
+    dbReader: DbReader[F],
+    deps: RuntimeCheckerDeps[F]
+  )(implicit F: Concurrent[F], logger: Logger[F], ev: Ask[F, TraceId]): CheckRunner[F, RuntimeWithWorkers] =
+    new CheckRunner[F, RuntimeWithWorkers] {
+      override def appName: String = resourceValidator.appName
+
+      override def configs = CheckRunnerConfigs(s"dataproc-workers", shouldAlert = true)
+
+      override def dependencies: CheckRunnerDeps[F] = deps.checkRunnerDeps
+
+      override def resourceToScan: fs2.Stream[F, RuntimeWithWorkers] = dbReader.getRuntimesWithWorkers
+
+      override def checkResource(runtime: RuntimeWithWorkers, isDryRun: Boolean)(
+        implicit ev: Ask[F, TraceId]
+      ): F[Option[RuntimeWithWorkers]] =
+        for {
+          clusterOpt <- deps.dataprocService
+            .getCluster(runtime.r.googleProject, regionName, DataprocClusterName(runtime.r.runtimeName))
+          runtime <- clusterOpt.fold[F[Option[RuntimeWithWorkers]]](F.pure(None)) { c =>
```

Comment:
Is there a better way to do this?

I don't think `traverse` or `traverse_` quite work because we want to go from Option[A] to F[Option[B]].

Reply:
`flatTraverse` should work similarly to in [this case](https://github.com/broadinstitute/leonardo-cron-jobs/blob/a89277b3d69835b27c5137a7c5f6391905254a8b/resource-validator/src/main/scala/com/broadinstitute/dsp/resourceValidator/StoppedRuntimeChecker.scala#L67) unless I'm missing something.